app/models/v2/report.rb

```diff
@@ -7,9 +7,64 @@ class Report < ApplicationRecord
     self.table_name = :reports
     self.primary_key = :id
 
+    SYSTEM_COUNT = lambda do
+      AN::NamedFunction.new(
+        'COUNT', [V2::System.arel_table[:id]]
+      ).filter(Pundit.policy_scope(User.current, V2::System).where_clause.ast)
+    end
+
+    COMPLIANT_SYSTEM_COUNT = lambda do
+      AN::NamedFunction.new(
+        'COUNT', [V2::System.arel_table[:id]]
+      ).filter(
+        Pundit.policy_scope(User.current, V2::System).where_clause.ast.and(
+          V2::TestResult.arel_table[:score].gteq(V2::Report.arel_table[:compliance_threshold])
+        )
+      )
+    end
+
+    UNSUPPORTED_SYSTEM_COUNT = lambda do
+      AN::NamedFunction.new(
+        'COUNT', [V2::System.arel_table[:id]]
+      ).filter(
+        Pundit.policy_scope(User.current, V2::System).where_clause.ast.and(
+          V2::TestResult.arel_table[:supported].eq('f')
```

Comment:
I'd rather go with `neq('t')` or its equivalent. The `supported` field might be NULL and pg uses a 3-state logic.